File: 6.sorting/lesson_6_distinct.py
def solution(A):
    A.sort()
    a = len(A)
    
    if(a == 0):
        return 0
    
    count = 1
    
    for i in range(1, a):
        if(A[i-1] != A[i]):
            count += 1

    return count

File: 6.sorting/lesson_6_distinct.js
function solution(A) {
    if(A.length == 0) {
        return 0
    }
    
    A = A.sort((a, b) => {return a - b})
    let distinctCount = 1
    let prevElement = A[0]
    for(let i = 1; i < A.length; i++) {
        if(A[i] != prevElement) {
            distinctCount++
            prevElement = A[i]
        }
    }
    return distinctCount
}
